src/app/libs/notification/components/notification.component.scss

```diff
@@ -0,0 +1,20 @@
+@mixin snack-style() {
+    text-align: center;
+    color: black;
+    font-size: 15px;
+}
+
+::ng-deep .snack-style-red {
+    @include snack-style();
+    background-color: red;
+}
+
+::ng-deep .snack-style-green {
+    @include snack-style();
+    background-color: green;
+}
+
+::ng-deep .snack-style-yellow {
+    @include snack-style();
+    background-color: yellow;
+}
```

Comment:
if everything is in `::ng-deep` here - just remove ng-deep and set ViewEncapsulation.None for the component. 
Read about this in Angular docs